src/index/visitor.rs

```diff
@@ -653,3 +536,296 @@ fn visit_data_type(
         _ => { /* unnamed datatypes are ignored */ }
     };
 }
+
+/// Internally we create a fat pointer struct for VLAs, which consists of a pointer to the passed array plus
+/// its dimensions, such that `ARRAY[*, *, *] OF INT` becomes
+/// ```ignore
+/// STRUCT
+///     ptr       : REF_TO ARRAY[*] OF INT;
+///     dimensions: ARRAY[0..2, 0..1] OF DINT;
+///                       ^^^^       --> dimension index
+///                             ^^^^ --> start- & end-offset
+/// END_STRUCT
+/// ```
+fn visit_variable_length_array(
+    bounds: &AstStatement,
+    referenced_type: &DataTypeDeclaration,
+    name: &str,
+    index: &mut Index,
+    type_declaration: &UserTypeDeclaration,
+    symbol_location_factory: &SymbolLocationFactory,
+) {
+    let ndims = match bounds {
+        AstStatement::VlaRangeStatement { .. } => 1,
+        AstStatement::ExpressionList { expressions, .. } => expressions.len(),
+        _ => unreachable!("not a bounds statement"),
+    };
+
+    let referenced_type = referenced_type.get_name().expect("named datatype").to_string();
```

Comment:
unfortunately I think this assumption is not correct.
you can inline-define a type and yes, it is strange to do this with a VLA, but you can.

example: declaring an enum
```st
PROGRAM
   VAR_INPUT
      normalArray: ARRAY[0..1] OF (RED, GREEN, BLUE);
      vlarArray: ARRAY[*] OF INT(0..1000); //e.g. subrange could make sense here?
   END_VAR 
END_PROGRAM
```